app/data_sources/zendesk.py

```diff
@@ -0,0 +1,192 @@
+import logging
+from abc import abstractmethod, ABC
+from datetime import datetime
+from enum import Enum
+from typing import Dict, List, Optional
+from data_source_api.exception import InvalidDataSourceConfig
+from data_source_api.basic_document import BasicDocument, DocumentType
+from indexing_queue import IndexingQueue
+import re
+import json
+import base64
+import requests
+from parsers.html import html_to_text
+from pydantic import BaseModel
+
+from db_engine import Session
+from schemas import DataSource
+
+#import http.client as http_client
+#http_client.HTTPConnection.debuglevel = 1
+
+class HTMLInputType(Enum):
+    TEXT = "text"
+    TEXTAREA = "textarea"
+    PASSWORD = "password"
+
+
+class ConfigField(BaseModel):
+    name: str
+    input_type: HTMLInputType = HTMLInputType.TEXT
+    label: Optional[str] = None
+    placeholder: Optional[str] = None
+
+    def __init__(self, **data):
+        name = data.get("name")
+        label = data.get("label") or name.title()
+        data["label"] = label
+        data["placeholder"] = data.get("placeholder") or label
+        super().__init__(**data)
+
+    class Config:
+        use_enum_values = True
+
+
+class ZendeskDataSource(ABC):
```

Comment:
why (ABC) and not (DataSource)?

Reply:
Because I'm bad at python inheritance and didn't read carefully enough. I'll fix tomorrow